src/tools/log-parser/README.md

```diff
@@ -0,0 +1,109 @@
+# `kata-log-parser`
+
+## Introduction
+
+`kata-log-parser` is a tool that combines logfiles generated by the various
+system components, sorts them by timestamp, and re-displays the log entries. A
+time delta is added to show how much time has elapsed between each log entry.
+
+The tool is also able to check the validity of all log records, can re-format the
+logs, and output them in a different format.
+
+For more information on the `kata-log-parser` tool, use the help command:
+
+```
+$ kata-log-parser --help
+```
+
+## Logfile requirements
+
+The tool reads logfiles in the [`logfmt`](https://brandur.org/logfmt) structured
+logging format. For example, a logfile created by the golang
+[Logrus](https://godoc.org/github.com/sirupsen/logrus) package.
+
+By default the tool requires that the following fields are defined for each log
+record:
+
+- Log level field (`level`): must be one of the Logrus `LogLevel` values
+  in string format (e.g. `debug`, `info`, `error`).
+
+- Name field (`name`): a single word that specifies the name of the
+  application that generates the log record (e.g. `kata-runtime`).
+
+- Process ID field (`pid`): the numeric process identifier for the process
+  that generates the log record.
+
+- Source field (`source`): a single word that specifies the name of a unique
+  part of the system (e.g. `runtime`).
+
+- Timestamp field (`time`): in [RFC3339](https://tools.ietf.org/html/rfc3339)
+  format and including a nanosecond value.
+
+Additional to the fields above, the tool also expects the following field:
+
+- Message field (`msg`): a textual message allowing log records to be
+  disambiguated.
+
+**Note:** These requirements can be ignored by using the `--ignore-missing-fields` flag
+
+## Component logfiles
+
+The primary logfiles the tool reads are:
+
+- The [runtime](https://github.com/kata-containers/kata-containers/tree/main/src/runtime) log.
+
+  This log also includes
+  [virtcontainers](https://github.com/containers/virtcontainers) log entries and
+  [agent](https://github.com/kata-containers/kata-containers/tree/main/src/agent) best effort logs unpacking (unless `--no-agent-unpack` is specified)
+
+## Usage
+
+To merge all logs:
+
+1. [Enable full debug](https://github.com/kata-containers/kata-containers/blob/main/docs/Developer-Guide.md#enable-full-debug).
+1. Clear the systemd journal (optional):
```

Comment:
I know I wrote this but, reading it again, I wonder if we should make it clearer that this should only be done on a test / dev system the user doesn't really care about as it's destructive? :smile:

A safer alternative is something like:

```bash
$ start_time=$(date '+%F %T')

# ... run stuff

$ sudo journalctl -q -o cat -a --since="$start_time" -t kata > ./kata.log

Reply:
added a comment for this option later